aries_vcx/tests/test_pgsql_plugin.rs

```diff
@@ -0,0 +1,117 @@
+
+mod pgwallet;
+
+fn _init_wallet(wallet_storage_config: &WalletStorageConfig) -> Result<(), String> {
+    info!("_init_wallet >>> wallet_storage_config:\n{}", serde_json::to_string(wallet_storage_config).unwrap());
+    match wallet_storage_config.xtype.as_ref() {
+        Some(wallet_type) => {
+            let (plugin_library_path_serialized,
+                plugin_init_function_serialized,
+                storage_config_serialized,
+                storage_credentials_serialized)
+                = serialize_storage_plugin_configuration(wallet_type,
+                                                         &wallet_storage_config.config,
+                                                         &wallet_storage_config.credentials,
+                                                         &wallet_storage_config.plugin_library_path,
+                                                         &wallet_storage_config.plugin_init_function)?;
+            let lib = load_storage_library(&plugin_library_path_serialized, &plugin_init_function_serialized)?;
+            if wallet_type == "postgres_storage" {
+                finish_loading_postgres(lib, &storage_config_serialized, &storage_credentials_serialized)?;
+            }
+            info!("Successfully loaded wallet plugin {}.", wallet_type);
+            Ok(())
+        }
+        None => {
+            info!("Using default builtin IndySDK wallets.");
+            Ok(())
+        }
+    }
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct WalletStorageConfig {
+    // Wallet storage type for agents wallets
+    #[serde(rename = "type")]
+    pub xtype: Option<String>,
+    // Optional to override default library path. Default value is determined based on value of
+    // xtype and OS
+    pub plugin_library_path: Option<String>,
+    // Optional to override default storage initialization function. Default value is  determined
+    // based on value of xtype and OS
+    pub plugin_init_function: Option<String>,
+    // Wallet storage config for agents wallets
+    pub config: Option<Value>,
+    // Wallet storage credentials for agents wallets
+    pub credentials: Option<Value>,
+}
+
+#[test]
+#[cfg(feature = "test_integration")]
+fn it_should_use_pgsql_wallet_to_provision_cloud_agent() {
+    let storage_config = r#"
+          {
+            "config": {
+              "url": "localhost:5432",
+              "max_connections" : 90,
+              "connection_timeout" : 30,
+              "wallet_scheme": "MultiWalletSingleTableSharedPool"
+            },
+            "credentials": {
+              "account": "postgres",
+              "password": "mysecretpassword",
+              "admin_acount": "postgres",
+              "admin_password": "mysecretpassword"
+            },
+            "type": "postgres_storage"
+          }"#;
+    let storage_config: WalletStorageConfig = serde_json::from_str(storage_config).unwrap();
```

Comment:
Why not bypass serde and just define the struct directly?

Reply:
`WalletStorageConfig` is using `Option<String>` for `config` and `credentials` fields, so it wouldn't look any nicer rn

Reply:
That could also be argument for defining the struct directly - the fact that some of the fields are `Option`s would not be hidden anymore. And not using dependencies when not necessary is a good rule of thumb. But that's just my opinion, I will leave it up to you.

Reply:
I did this in addition to whole bunch of additional refactoring and simplification.